src/System.Net.Http/tests/FunctionalTests/HttpClientMiniStressTest.cs

```diff
@@ -0,0 +1,120 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.IO;
+using System.Net.Sockets;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace System.Net.Http.Functional.Tests
+{
+    public class HttpClientMiniStress
+    {
+        public static IEnumerable<object[]> StressOptions()
+        {
+            foreach (int numRequests in new[] { 5000 })
+                foreach (int dop in new[] { 1, 32 })
+                    foreach (var completionoption in new[] { HttpCompletionOption.ResponseContentRead, HttpCompletionOption.ResponseHeadersRead })
+                        yield return new object[] { numRequests, dop, completionoption };
+        }
+
+        [OuterLoop]
+        [Theory]
+        [MemberData(nameof(StressOptions))]
+        public void SingleClient_ManyRequests(int numRequests, int dop, HttpCompletionOption completionOption)
+        {
+            string responseText = CreateResponse("abcdefghijklmnopqrstuvwxyz");
+            using (var client = new HttpClient())
+            {
+                Parallel.For(0, numRequests, new ParallelOptions { MaxDegreeOfParallelism = dop }, _ =>
```

Comment:
I wonder if you should use explicit threads (via LongRunningTask) for these.  With Parallel.For on most machines you're effectively going to be limited to the number of logical processors; since these are relatively long-running operations, the ThreadPool will likely take a while to increase the thread count much.